files/default/test-neutron-ha-tool.py

```diff
@@ -0,0 +1,392 @@
+    def test_default_strategy_is_least_busy_agent_picker(self):
+        configure_with('')
+
+        self.assertIsInstance(
+            ha_tool.Configuration.agent_picker,
+            ha_tool.LeastBusyAgentPicker
+        )
+
+    def test_neutron_client_injected_to_least_busy_agent_picker(self):
+        configure_with('', 'qclient')
+
+        self.assertEqual(
+            'qclient',
+            ha_tool.Configuration.agent_picker.qclient
+        )
+
+    def test_selecting_random_agent_picker(self):
+        configure_with('--agent-selection-mode random')
+
+        self.assertIsInstance(
+            ha_tool.Configuration.agent_picker,
+            ha_tool.RandomAgentPicker
+        )
+
+    def test_explicit_agent_specified(self):
+        configure_with('--target-agent xyz')
+
+        self.assertIsInstance(
+            ha_tool.Configuration.agent_picker,
+            ha_tool.SingleAgentPicker
+        )
+        self.assertEqual(
+            'xyz',
+            ha_tool.Configuration.agent_picker.agent_selection_value
+        )
+
+
+class TestRouterFilterConfiguration(unittest.TestCase):
+    def test_null_router_filter_is_the_default(self):
+        configure_with('')
+
+        self.assertIsInstance(
+            ha_tool.Configuration.router_filter,
+            ha_tool.NullRouterFilter
+        )
+
+    def test_set_router_list(self):
+        configure_with('--router-list-file routers.lst')
+
+        self.assertIsInstance(
+            ha_tool.Configuration.router_filter,
+            ha_tool.ListFileBasedRouterFilter
+        )
+        self.assertEqual(
+            'routers.lst',
+            ha_tool.Configuration.router_filter.router_list_file
+        )
+
+
+class TestListFileBasedRouterFilter(unittest.TestCase):
+    def test_if_file_does_not_exist_system_error_raised(self):
+        router_filter = ha_tool.ListFileBasedRouterFilter('nonexisting')
+
+        with self.assertRaises(IOError) as ctx:
+            router_filter.load()
+
+    def test_loading_router_list_file(self):
+        with tempfile.NamedTemporaryFile() as list_file:
+            list_file.write('id-1\n')
+            list_file.write('\n')
+            list_file.write('     \n')
+            list_file.write('\t\t\n')
+            list_file.write('id-2')
+            list_file.seek(0)
+            
+            router_filter = ha_tool.ListFileBasedRouterFilter(list_file.name)
+            router_filter.load()
+
+        self.assertEqual(
+            ['id-1', 'id-2'],
+            router_filter.router_list
+        )
+
+    def test_filtering_empty_file(self):
+        router_filter = ha_tool.ListFileBasedRouterFilter(None)
+        router_filter.router_list = []
+
+        self.assertEqual(
+            [],
+            router_filter.filter_routers(['a', 'b'])
+        )
+
+    def test_filtering_some_routers(self):
+        router_filter = ha_tool.ListFileBasedRouterFilter(None)
+        router_filter.router_list = ['a']
```

Comment:
but `filter_routers` is just returning `self.router_list` which was set to the value on the same test. This is just checking that assignment and return functions works as expected but contains no real logic that I can see. In this test you can pass any argument to `filter_routers` and it will always return the assignment of `["a"]` that you did on line 382.